manual_flagging/flagging.py

```diff
@@ -242,73 +237,33 @@ def create_bins_and_labels(input_list):
         dfs_to_rolling_window[key] = {"df": df_filtered}  # Store the filtered DataFrame
 
         # Extract the specific housing type configuration
-        housing_type_config = inputs["stat_groups"]["current"][f"tri{tri}"][
-            housing_type
-        ]
-
-        # Update market/tri configurations
-        dfs_to_rolling_window[key]["columns"] = housing_type_config["columns"]
+        housing_type_config = inputs["stat_groups"][f"tri{tri}"][housing_type]
+
+        # Perform column transformations
+        columns = housing_type_config["columns"]
+        transformed_columns = []
+
+        for col in columns:
+            if isinstance(col, dict) and "column" in col:
+                if "bins" in col:
+                    bins, labels = create_bins_and_labels(col["bins"])
+                    new_col_name = f"{col['column']}_bin"
+                    df_filtered[new_col_name] = pd.cut(
+                        df_filtered[col["column"]], bins=bins, labels=labels
+                    )
+                    transformed_columns.append(new_col_name)
+            else:
+                transformed_columns.append(col)
+
+        dfs_to_rolling_window[key]["columns"] = transformed_columns
+
+        # Add rest of config information
```

Comment:
This loop enables take the structure that exists in the `stat_groups` yaml object:
```yaml
      columns:
        - "rolling_window"
        - "geography_split"
        - column: "char_bldg_age"
          bins: [1200, 2400]
        - column: "char_bldg_sf"
          bins: [40]
```
and pre-processes the data such the relevant features are created and correctly processed through the script.